problemtools/verifyproblem.py

```diff
@@ -549,6 +583,11 @@ def check(self, args):
         elif self._data['license'] == 'unknown':
             self.warning("License is 'unknown'")
 
+        if self._data['grading']['testgroups'] not in [True, False]:
+            self.error("Invalid value for grading.testgroups: %s" % self._data['grading']['testgroups'])
+        elif self._data['grading']['testgroups'] and self._data['type'] == 'pass-fail':
+            self.error("Test groups are only supported for scoring problems, this is a pass-fail problem")
```

Comment:
The `grading` map can only be used with scoring problems, so this check should be one level higher.

Reply:
Ok, fair enough...